### PR TITLE
Better: Add Instagram Messaging capabilities and specificities to API doc. RD-19460

### PR DESCRIPTION
**Pull request by ytuauden**
Add Instagram Messaging capabilities and specificities to API doc

### CODE REVIEW

File: docs/interactions/structured-messages/index.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Engage Digital provides an omnichannel API to create structured messages. Since
   | **Twitter Direct Messages** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | N/A | N/A | N/A |  N/A |
   | **Viber** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | N/A | N/A |  N/A |
   | **WhatsApp** | **<span style="color:green">Yes</span>** | N/A | **<span style="color:green">Yes</span>** | N/A | N/A | N/A |  N/A |
+  | **Instagram Messaging** | **<span style="color:green">Yes</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | N/A | N/A |  N/A |
```
[the remark this answers is not in the document]
Comment (CedricBm) [reply docs/interactions/structured-messages/index.md]: heh, of course when you do the modification, Instagram releases these template, so you can put back the "Not yet"

cf RD-19647

[the remark this answers is not in the document]
Comment (ytuauden) [reply docs/interactions/structured-messages/index.md]: ahah yes I was confused as I just removed the Not Yet and then I saw [this](https://developers.facebook.com/docs/messenger-platform/instagram/features/generic-template/) and [this](https://developers.facebook.com/docs/messenger-platform/instagram/features/product-template/#carousel) so it seems that carousel and templates are now supported by Instagram graph API

[the remark this answers is not in the document]
Comment (CedricBm) [reply docs/interactions/structured-messages/index.md]: Actually we'll also support rich link by converting to a template with a single item being the url. So you can also put "Not yet" on the rich link

File: docs/interactions/structured-messages/select.md [hunk 2]
```diff
@@ -165,6 +165,7 @@ Primary parameters are used by default, however, some parameters are unique or o
 | **`items.title`** | String | The title of the item. *Truncated to 20 characters.* |
 | **`items.attachment_id`** | String | **Optional**. Existing attachment id used to decorate the item in the list. Supports private attachments. [Upload attachments](../../../basics/uploads) for you own custom images. |
 | **`items.attachment_fallback_id`** | String | **Optional**. Fallback in case the attachment related to the attachment_id doesn’t meet the source requirements. Must be public. Only jpg, jpeg, png formats. Maximum size of 5 MB. [Upload attachments](../../../basics/uploads) for you own custom images. |
+| **`items.payload`** | String | **Optional**. The item payload field. Limited to 200 characters. Automatically gets populated as a random hex if blank. |
```
Comment (CedricBm) [inline docs/interactions/structured-messages/select.md:168]: 1. Why did you add back it's optional and it's the item payload field? It's already said in the generic table, here it's about specificities
2. It's not limited to 200 chars, only WA is.
3. Can you also handle the other strmes type? Like template and carousel.

Comment (ytuauden) [reply docs/interactions/structured-messages/select.md:168]: 1. I just copied the text from WA and wanted it to be consistant with the other sources
2. OK didn't know about that thank you
3. yes OK

**Comment by CedricBm** [comment]
Looks good to me.
